lib/media_types/scheme.rb

```diff
@@ -385,24 +396,57 @@ def inspect(indentation = 0)
     end
 
     def assert_pass(fixture)
-      json = JSON.parse(fixture, { symbolize_names: true })
-
-      validate(json)
+      caller = caller_locations[1]
+      object_to_store = {
+        fixture: fixture,
+        expect_to_pass: true,
+        caller: caller
+        }
+      @fixtures << object_to_store
     end
-    
+
     def assert_fail(fixture)
-      json = JSON.parse(fixture, { symbolize_names: true })
+      caller = caller_locations[1]
+      object_to_store = {
+        fixture: fixture,
+        expect_to_pass: false,
+        caller: caller
+        }
+      @fixtures << object_to_store
+    end
+
+    def execute_assertions(media_type_class)
+      errors = []
+      @fixtures.each do |object|
+        json = JSON.parse(object[:fixture], { symbolize_names: true })
+        caller = object[:caller]
+        output =  object[:expect_to_pass] ? process_assert_pass(json, caller) : process_assert_fail(json, caller)
+        errors << output unless output.nil?
+      end
+      raise AssertionError, errors.map { |error| error.msg } unless errors.empty?
+      @asserted_sane = true
+    end
 
+    def process_assert_fail(json, caller)
+      expectation_met = false
       begin
         validate(json)
       rescue MediaTypes::Scheme::ValidationError
-        return
+        expectation_met = true
       end
-      raise AssertionError
+      MediaTypeValidationError.new(json, caller) if expectation_met == false
     end
 
-    private
+    def process_assert_pass(json, caller)
+      begin
+        validate(json)
+      rescue => exception
+        error = exception.message + " at #{caller.path + ':' + caller.lineno.to_s}"
+      end
+      error
```

Comment:
```suggestion
        exception.message + " at #{caller.path + ':' + caller.lineno.to_s}"
      end
```

Doest this not work?

Reply:
I am trying to do one of the refacorings requested

and I cannot see why

   def process_assert_pass(json, caller)
      validate(json)
    rescue StandardError => e
      error = e.message + "\n#{caller.path + ':' + caller.lineno.to_s}"
    end
sometimes returns true, which is not what I want

while

def process_assert_pass(json, caller)
      error = nil
      begin
        validate(json)
      rescue StandardError => e
        error = e.message + "\n#{caller.path + ':' + caller.lineno.to_s}"
      end
      error
    end
Does not

Do either of you see what the error I am making could be?

this is also the case if you remove the error on line 8 of the second example

Reply:
I went through this process when I first wrote the method, as I also initially thought the suggested way would provide the same result